test/rule-syntax-check.py

```diff
@@ -28,10 +28,11 @@
 if not rules_files:
     sys.exit('Specify files to test as arguments')
 
-no_args_tests = re.compile(r'(ACTION|DEVPATH|KERNELS?|NAME|SYMLINK|SUBSYSTEMS?|DRIVERS?|TAG|RESULT|TEST)\s*(?:=|!)=\s*"([^"]*)"$')
-args_tests = re.compile(r'(ATTRS?|ENV|TEST){([a-zA-Z0-9/_.*%-]+)}\s*(?:=|!)=\s*"([^"]*)"$')
-no_args_assign = re.compile(r'(NAME|SYMLINK|OWNER|GROUP|MODE|TAG|PROGRAM|RUN|LABEL|GOTO|OPTIONS|IMPORT)\s*(?:\+=|:=|=)\s*"([^"]*)"$')
-args_assign = re.compile(r'(ATTR|ENV|IMPORT|RUN){([a-zA-Z0-9/_.*%-]+)}\s*(=|\+=)\s*"([^"]*)"$')
+quoted_string_re = r'"(?:[^\\"]|\\.)*"'
```

Comment:
@filbranden, thank you.

It took me a while to remember why I had used a little more complicated regular expression :-) `udev` accepts weird strings like `PROGRAM="\\""` that can't be properly detected by any regular expression expecting the 
normal quoted string. On the other hand, it is probably better to reject such strings because they simply shouldn't be used in anything written by people.

Reply:
Oh yes indeed you are correct. I wonder if this should be "fixed" in udev by making it translate `\\` -> `\` (same as it translates `\"` -> `"`) while leaving any other `\x` sequences unchanged...

Right now, as it is, udev rules can't really produce a single backslash followed by a double-quote character. Making `\\` work would fix that.

Reply:
See #8301.